app/assets/stylesheets/govuk_publishing_components/components/_heading.scss

```diff
@@ -45,3 +45,25 @@
 .gem-c-heading--inverse {
   color: govuk-colour("white");
 }
+
+// stylelint-disable declaration-no-important
+@include govuk-media-query($media-type: print) {
+
+  // Override margins and colors
+  .gem-c-heading {
+    margin: 0 !important;
+    color: $govuk-print-text-colour !important;
+    border-color: $govuk-print-text-colour !important;
+  }
+
+  // Remove extra padding to reduce whitespace, except for headings with borders
+  .gem-c-heading--padding:not([class*="--border-top"]) {
+    padding: 0 !important;
+  }
+
+  // Remove lower padding for headings with borders
+  .gem-c-heading--padding[class*="--border-top"] {
```

Comment:
Appreciate that this is slightly complex to restyle given the various colliding classes, but I think this whole section can be written slightly simpler as:

```CSS
// stylelint-disable declaration-no-important
@include govuk-media-query($media-type: print) {
  // Override margins and colors
  .gem-c-heading {
    margin: 0 !important;
    color: $govuk-print-text-colour !important;
    border-color: $govuk-print-text-colour !important;
    padding: 0;
  }

  .gem-c-heading--padding[class*="--border-top"] {
    padding-top: govuk-spacing(3);
  }
}
// stylelint-enable declaration-no-important

```

Reply:
Aha, yeah that's cleaner, thanks for taking the time to improve it.